multiplatform-markdown-renderer/src/commonMain/kotlin/com/mikepenz/markdown/compose/Markdown.kt

```diff
@@ -0,0 +1,107 @@
+package com.mikepenz.markdown.compose
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Modifier
+import com.mikepenz.markdown.compose.elements.MarkdownBlockQuote
+import com.mikepenz.markdown.compose.elements.MarkdownBulletList
+import com.mikepenz.markdown.compose.elements.MarkdownCodeBlock
+import com.mikepenz.markdown.compose.elements.MarkdownCodeFence
+import com.mikepenz.markdown.compose.elements.MarkdownHeader
+import com.mikepenz.markdown.compose.elements.MarkdownImage
+import com.mikepenz.markdown.compose.elements.MarkdownOrderedList
+import com.mikepenz.markdown.compose.elements.MarkdownParagraph
+import com.mikepenz.markdown.compose.elements.MarkdownText
+import com.mikepenz.markdown.model.*
+import org.intellij.markdown.MarkdownElementTypes
+import org.intellij.markdown.MarkdownTokenTypes
+import org.intellij.markdown.ast.ASTNode
+import org.intellij.markdown.ast.findChildOfType
+import org.intellij.markdown.ast.getTextInNode
+import org.intellij.markdown.flavours.MarkdownFlavourDescriptor
+import org.intellij.markdown.flavours.gfm.GFMFlavourDescriptor
+import org.intellij.markdown.parser.MarkdownParser
+
+@Composable
+fun Markdown(
+    content: String,
+    colors: MarkdownColors,
+    typography: MarkdownTypography,
+    padding: MarkdownPadding,
+    modifier: Modifier = Modifier.fillMaxSize(),
+    flavour: MarkdownFlavourDescriptor = GFMFlavourDescriptor()
+) {
+    CompositionLocalProvider(
+        LocalReferenceLinkHandler provides object : ReferenceLinkHandler {
+            private val stored = mutableMapOf<String, String?>()
+            override fun store(label: String, destination: String?) {
+                stored[label] = destination
+            }
+            override fun find(label: String): String {
+                return stored[label] ?: label
+            }
+        },
+        LocalBulletListHandler provides object : BulletHandler {
+            override fun transform(bullet: CharSequence?): String {
+                return "• "
+            }
+        },
+        LocalOrderedListHandler provides object : BulletHandler {
+            override fun transform(bullet: CharSequence?): String {
+                return "$bullet "
+            }
+        },
+        LocalMarkdownPadding provides padding,
+        LocalMarkdownColors provides colors,
+        LocalMarkdownTypography provides typography,
+    ) {
```

Comment:
Could you please explain the choice of setting the defaults for the `compositionLocalProvider` as part of the `Markdown` class? 
Instead of having the default implementation in its own place? Ultimately the `Markdown` would not require to know about the specific implementation?